src/useScroll.ts

```diff
@@ -17,14 +21,35 @@ const useScroll = (ref: RefObject<HTMLElement>): State => {
   const [state, setState] = useRafState<State>({
     x: 0,
     y: 0,
+    direction: {
+      horizontal: 'none',
+      vertical: 'none'
+    }
+    
   });
 
   useEffect(() => {
     const handler = () => {
       if (ref.current) {
+        let horizontal:'up' | 'down' | 'none' = 'none',
+            vertical:'left' | 'right' | 'none' = 'none'; 
+
+        if(ref.current.scrollTop) {
+          if(state.y > ref.current.scrollTop) horizontal = 'down';
+          else if(state.y < ref.current.scrollTop) horizontal = 'up';
+        }
+  
+        if(ref.current.scrollLeft) {
+          if(state.x > ref.current.scrollLeft) vertical = 'left';
+          else if(state.x < ref.current.scrollLeft) vertical = 'right';
+        }
+
         setState({
           x: ref.current.scrollLeft,
           y: ref.current.scrollTop,
+          direction: {
+            horizontal, vertical
+          }
         });
       }
```

Comment:
I have a suggestion to simplify and un-nest the code, exit early. I also aligned the formatting (trailing commas, space after `if`).

```suggestion
      if (!ref.current) {
        return;
      }
      
      const { scrollLeft, scrollTop } = ref.current;

      let horizontal: 'up' | 'down' | 'none' = 'none';
      let vertical: 'left' | 'right' | 'none' = 'none'; 
      
      if (state.y > scrollTop) horizontal = 'down';
      else if (state.y < scrollTop) horizontal = 'up';
  
      if (state.x > scrollLeft) vertical = 'left';
      else if (state.x < scrollLeft) vertical = 'right';
      
      setState({
        x: scrollLeft,
        y: scrollTop,
        direction: {
          horizontal, 
          vertical,
        },
      });
```

You could also rename `scrollLeft` and `scrollTop` to `y` and `x` respectively during destructuring, but I didn't make that change.

Reply:
Unfortunately, this package looks no longer to be maintained :(